Master branch doesn't build. GCC incompatibility with clang.

Stack Overflow says:  _"In gcc stl system_clock duration is nanoseconds, but in clang it is microseconds."_

$ make
Scanning dependencies of target opendht
[  4%] Building CXX object CMakeFiles/opendht.dir/src/utils.cpp.o
/Users/au/Projects/opendht/src/utils.cpp:29:49: error: no viable conversion from
      **'time_point<[...], duration<[...], ratio<[...], 1000000000>>>' to 'const**
      **time_point<[...], duration<[...], ratio<[...], 1000000>>>'**
    return std::chrono::system_clock::to_time_t(std::chrono::system_clock::now() + 
                                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/au/bin/clang+llvm-3.5.0-macosx-apple-darwin/bin/../include/c++/v1/chrono:750:29: note: 
      candidate constructor (the implicit copy constructor) not viable: no known conversion from
      'time_point<std::__1::chrono::system_clock, typename common_type<duration<long long,
      ratio<1, 1000000> >, duration<long long, ratio<1, 1000000000> > >::type>' to 'const
      std::__1::chrono::time_point<std::__1::chrono::system_clock,
      std::__1::chrono::duration<long long, std::__1::ratio<1, 1000000> > > &' for 1st argument
class _LIBCPP_TYPE_VIS_ONLY time_point
                            ^
/Users/au/bin/clang+llvm-3.5.0-macosx-apple-darwin/bin/../include/c++/v1/chrono:750:29: note: 
      candidate constructor (the implicit move constructor) not viable: no known conversion from
      'time_point<std::__1::chrono::system_clock, typename common_type<duration<long long,
      ratio<1, 1000000> >, duration<long long, ratio<1, 1000000000> > >::type>' to
      'std::__1::chrono::time_point<std::__1::chrono::system_clock,
      std::__1::chrono::duration<long long, std::__1::ratio<1, 1000000> > > &&' for 1st argument
class _LIBCPP_TYPE_VIS_ONLY time_point
                            ^
/Users/au/bin/clang+llvm-3.5.0-macosx-apple-darwin/bin/../include/c++/v1/chrono:772:13: note: 
      candidate template ignored: disabled by 'enable_if' [with _Duration2 =
      std::__1::chrono::duration<long long, std::__1::ratio<1, 1000000000> >]
            is_convertible<_Duration2, duration>::value
            ^
/Users/au/bin/clang+llvm-3.5.0-macosx-apple-darwin/bin/../include/c++/v1/chrono:925:53: note: 
      passing argument to parameter '__t' here
    static time_t     to_time_t  (const time_point& __t) _NOEXCEPT;
                                                    ^
1 error generated.
make[2]: *** [CMakeFiles/opendht.dir/src/utils.cpp.o] Error 1
make[1]: *** [CMakeFiles/opendht.dir/all] Error 2
make: *** [all] Error 2

--- a/src/utils.cpp
+++ b/src/utils.cpp
@@ -26,8 +26,9 @@ time_point from_time_t(std::time_t t) {
 }
 
 std::time_t to_time_t(time_point t) {
-    return std::chrono::system_clock::to_time_t(std::chrono::system_clock::now() + 
-            (t - clock::now()));
+    return std::chrono::system_clock::to_time_t(
+            std::chrono::system_clock::now() +
+            std::chrono::duration_cast<std::chrono::system_clock::duration>(t - clock::now()));
 }
 
 Blob

